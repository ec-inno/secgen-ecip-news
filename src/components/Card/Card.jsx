import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'gatsby';
import { useTranslation } from 'react-i18next';

const Card = ({
  title,
  status,
  background,
  href,
  totalSupporters,
  supportLink,
}) => {
  const { t } = useTranslation();

  return (
    <article className="ecl-card">
      <header className="ecl-card__header">
        <span className="ecl-tag ecl-u-f-r eci-tag">{status}</span>
        <div
          role="img"
          className="ecl-card__image"
          aria-label={title}
          style={{
            backgroundImage: `url('${background}')`,
          }}
        ></div>
        <h1 className="ecl-card__title">
          <Link
            to={href}
            className="ecl-link ecl-link--standalone"
            aria-label={title}
            level="1"
          >
            {title}
          </Link>
        </h1>
      </header>
      <section className="ecl-card__body">
        <p className="ecl-u-type-paragraph ecl-u-mv-none">
          {totalSupporters} {t('supporters')}
        </p>
        {supportLink && (
          <ul className="ecl-card__link-container">
            <li className="ecl-card__link-item">
              <a
                target="_blank"
                href={supportLink}
                className="ecl-card__link ecl-link ecl-link--standalone"
              >
                {t('Support this initiative')}
              </a>
            </li>
          </ul>
        )}
      </section>
    </article>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  totalSupporters: PropTypes.number,
  supportLink: PropTypes.string,
  background: PropTypes.string,
};

export default Card;
