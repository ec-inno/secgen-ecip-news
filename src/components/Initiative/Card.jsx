import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import axios from 'axios';
import { Link } from 'gatsby';
import { useTranslation } from 'react-i18next';

import defaultImage from '../assets/images/default-image.png';
import formatStatus from '../../utils/formatStatus';

const Card = ({ title, status, logo, href, totalSupporters, supportLink }) => {
  const { t } = useTranslation();
  const [logoData, setLogoData] = useState('');

  const { GATSBY_INITIATIVES_API: api } = process.env;
  const hasLogo = logo && logo.id && logo.mimeType;

  useEffect(() => {
    if (hasLogo) {
      axios
        .get(`${api}/register/logo/${logo.id}`, {
          responseType: 'arraybuffer',
        })
        .then(r => {
          const data = Buffer.from(r.data, 'binary').toString('base64');
          setLogoData(data);
        })
        .catch(e => {
          console.error(`Failed to fetch logo ${logo.id}`, e);
        });
    }
  }, []);

  const background = hasLogo
    ? `data:${logo.mimeType};base64,${logoData}`
    : defaultImage;

  return (
    <article className="ecl-card">
      <header className="ecl-card__header">
        <span className="ecl-tag ecl-u-f-r eci-tag">
          {formatStatus(status)}
        </span>
        <div
          role="img"
          className="ecl-card__image"
          aria-label={title}
          style={{
            backgroundImage: `url('${background}')`,
          }}
        ></div>
        <h1 className="ecl-card__title">
          <Link
            to={href}
            className="ecl-link ecl-link--standalone"
            aria-label={title}
            level="1"
          >
            {title}
          </Link>
        </h1>
      </header>
      <section className="ecl-card__body">
        <p className="ecl-u-type-paragraph ecl-u-mv-none">
          {totalSupporters} {t('supporters')}
        </p>
        {supportLink && (
          <ul className="ecl-card__link-container">
            <li className="ecl-card__link-item">
              <a
                target="_blank"
                href={supportLink}
                className="ecl-card__link ecl-link ecl-link--standalone"
              >
                {t('Support this initiative')}
              </a>
            </li>
          </ul>
        )}
      </section>
    </article>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  totalSupporters: PropTypes.string,
  supportLink: PropTypes.string,
  logo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    mimeType: PropTypes.string,
    size: PropTypes.number,
  }),
};

export default Card;
