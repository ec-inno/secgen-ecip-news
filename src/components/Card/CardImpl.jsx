import React from 'react';
import PropTypes from 'prop-types';

import upperCaseFirstChar from '@eci/utils/upperCaseFirstChar';
import { useLogoApi } from '@eci/utils/useLogoApi';

import Card from './Card';
import defaultImage from '../assets/images/default-image.png';

const CardImpl = ({
  title,
  status,
  logo,
  href,
  totalSupporters,
  supportLink,
}) => {
  if (logo && logo.id && logo.mimeType) {
    const { data } = useLogoApi(logo.id);

    return (
      <Card
        title={title}
        href={href}
        status={upperCaseFirstChar(status)}
        totalSupporters={totalSupporters}
        supportLink={supportLink}
        background={`data:${logo.mimeType};base64,${data}`}
      />
    );
  }

  return (
    <Card
      title={title}
      href={href}
      status={upperCaseFirstChar(status)}
      totalSupporters={totalSupporters}
      supportLink={supportLink}
      background={defaultImage}
    />
  );
};

CardImpl.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  totalSupporters: PropTypes.number,
  supportLink: PropTypes.string,
  logo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    mimeType: PropTypes.string,
    size: PropTypes.number,
  }),
};

export default CardImpl;
