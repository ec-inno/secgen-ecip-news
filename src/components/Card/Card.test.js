import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays support link when target is provided', () => {
    const tree = renderer.create(<Card supportLink="#" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
