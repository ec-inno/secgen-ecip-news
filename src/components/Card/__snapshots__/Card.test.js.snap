// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card displays support link when target is provided 1`] = `
<article
  className="ecl-card"
>
  <header
    className="ecl-card__header"
  >
    <span
      className="ecl-tag ecl-u-f-r eci-tag"
    />
    <div
      className="ecl-card__image"
      role="img"
      style={
        Object {
          "backgroundImage": "url('undefined')",
        }
      }
    />
    <h1
      className="ecl-card__title"
    >
      <a
        className="ecl-link ecl-link--standalone"
        level="1"
      />
    </h1>
  </header>
  <section
    className="ecl-card__body"
  >
    <p
      className="ecl-u-type-paragraph ecl-u-mv-none"
    >
       
      supporters
    </p>
    <ul
      className="ecl-card__link-container"
    >
      <li
        className="ecl-card__link-item"
      >
        <a
          className="ecl-card__link ecl-link ecl-link--standalone"
          href="#"
          target="_blank"
        >
          Support this initiative
        </a>
      </li>
    </ul>
  </section>
</article>
`;

exports[`Card renders correctly 1`] = `
<article
  className="ecl-card"
>
  <header
    className="ecl-card__header"
  >
    <span
      className="ecl-tag ecl-u-f-r eci-tag"
    />
    <div
      className="ecl-card__image"
      role="img"
      style={
        Object {
          "backgroundImage": "url('undefined')",
        }
      }
    />
    <h1
      className="ecl-card__title"
    >
      <a
        className="ecl-link ecl-link--standalone"
        level="1"
      />
    </h1>
  </header>
  <section
    className="ecl-card__body"
  >
    <p
      className="ecl-u-type-paragraph ecl-u-mv-none"
    >
       
      supporters
    </p>
  </section>
</article>
`;
